src/Popover/Popover.tsx

```diff
@@ -315,13 +314,18 @@ export class Popover extends React.Component<Popover.Props, State> {
     const hiddenPopover = this.getHiddenPopover();
     const { context } = this.getPopoverProps();
     const { ContextWrapper } = this;
-    ReactDOM.render(<ContextWrapper context={context}>{hiddenPopover}</ContextWrapper>, this.containerNode);
 
-    // add pointer to popover node
-    this.popoverNode = this.containerNode.children[0];
+    // render Popover and save popover size in "onMount" cb (visible popover will be rendered in componentDidUpdate)
```

Comment:
did you read this https://github.com/facebook/react/issues/10309#issuecomment-318434635 ?
Seems slightly simpler (but same thing at an higher level)